src/backend/opengl/PersistentPipelineStateGL.cpp

```diff
@@ -0,0 +1,169 @@
+        }
+    }
+
+    void PersistentPipelineState::ApplyStencilNow() {
+        if (IsDirty(STENCIL_ENABLED)) {
+            if (state.stencilEnabled) {
+                glEnable(GL_STENCIL_TEST);
+            } else {
+                glDisable(GL_STENCIL_TEST);
+            }
+            CleanField(STENCIL_ENABLED);
+        }
+
+        if (IsDirty(STENCIL_BACK_STENCIL_FAIL) ||
+            IsDirty(STENCIL_BACK_DEPTH_FAIL) ||
+            IsDirty(STENCIL_BACK_DEPTH_STENCIL_PASS)) {
+
+            glStencilOpSeparate(GL_BACK,
+                OpenGLStencilOperation(state.stencilInfo.back.stencilFail),
+                OpenGLStencilOperation(state.stencilInfo.back.depthFail),
+                OpenGLStencilOperation(state.stencilInfo.back.depthStencilPass)
+            );
+
+            CleanField(STENCIL_BACK_STENCIL_FAIL);
+            CleanField(STENCIL_BACK_DEPTH_FAIL);
+            CleanField(STENCIL_BACK_DEPTH_STENCIL_PASS);
+        }
+        if (IsDirty(STENCIL_BACK_COMPARE_FUNCTION) ||
+            IsDirty(STENCIL_REFERENCE) ||
+            IsDirty(STENCIL_BACK_MASK)) {
+
+            glStencilFuncSeparate(GL_BACK,
+                OpenGLCompareFunction(state.stencilInfo.back.compareFunction),
+                state.stencilReference,
+                state.stencilInfo.back.mask
+            );
+            if (IsDirty(STENCIL_BACK_MASK)) {
+                glStencilMaskSeparate(GL_BACK, state.stencilInfo.back.mask);
+            }
+
+            CleanField(STENCIL_BACK_COMPARE_FUNCTION);
+            CleanField(STENCIL_BACK_MASK);
+        }
+
+        if (IsDirty(STENCIL_FRONT_STENCIL_FAIL) ||
+            IsDirty(STENCIL_FRONT_DEPTH_FAIL) ||
+            IsDirty(STENCIL_FRONT_DEPTH_STENCIL_PASS)) {
+
+            glStencilOpSeparate(GL_FRONT,
+                OpenGLStencilOperation(state.stencilInfo.front.stencilFail),
+                OpenGLStencilOperation(state.stencilInfo.front.depthFail),
+                OpenGLStencilOperation(state.stencilInfo.front.depthStencilPass)
+            );
+
+            CleanField(STENCIL_FRONT_STENCIL_FAIL);
+            CleanField(STENCIL_FRONT_DEPTH_FAIL);
+            CleanField(STENCIL_FRONT_DEPTH_STENCIL_PASS);
+        }
+        if (IsDirty(STENCIL_FRONT_COMPARE_FUNCTION) ||
+            IsDirty(STENCIL_REFERENCE) ||
+            IsDirty(STENCIL_FRONT_MASK)) {
+
+            glStencilFuncSeparate(GL_FRONT,
+                OpenGLCompareFunction(state.stencilInfo.front.compareFunction),
+                state.stencilReference,
+                state.stencilInfo.front.mask
+            );
+            if (IsDirty(STENCIL_FRONT_MASK)) {
+                glStencilMaskSeparate(GL_FRONT, state.stencilInfo.front.mask);
+            }
+
+            CleanField(STENCIL_FRONT_COMPARE_FUNCTION);
+            CleanField(STENCIL_FRONT_MASK);
+        }
+
+        CleanField(STENCIL_REFERENCE); // clean this last because its used for both the back and front functions
+    }
+
+}
+}
```

Comment:
I think you are missing a newline at the end of the file (which is actually against the C standard IIRC)